### PR TITLE
updated release.yml + python-test.yml to include more scoped permissions as per issue #337

### PR DESCRIPTION
**Pull request by t1nfoil**
We've updated the workflow files with the more granular scoped permissions, please test and approve as needed.

cc @amaranthjinn

### CODE REVIEW

**Review by gvwilson — APPROVED** [review]

**Comment by amaranthjinn** [comment]
6.5/10 

5 / 10  | Branch-Protection    Warn:  'stale review dismissal' is                                                                                count is 1 on branch 'main' Warn: codeowners review is                                                                           required - but no codeowners file found in repo Warn: 'last                                                                        push approval' is disable on  branch 'main' Warn: no status                                                                                   checks found to merge onto branch 'main'

0/10 signed release

3/10 vulnerabilities -7 detected. Note, osv scanner is more reliable for this.

**Pull request merged**
